lib/python/qmk/cli/generate/version_h.py

```diff
@@ -29,23 +29,30 @@ def generate_version_h(cli):
         current_time = strftime(TIME_FMT)
 
     if cli.args.skip_git:
+        git_dirty = False
+        git_commit = "NA"
         git_version = "NA"
         chibios_version = "NA"
         chibios_contrib_version = "NA"
     else:
+        git_dirty = git_is_dirty()
+        git_commit = git_get_commit_hash()
         git_version = git_get_version() or current_time
         chibios_version = git_get_version("chibios", "os") or current_time
         chibios_contrib_version = git_get_version("chibios-contrib", "os") or current_time
 
     # Build the version.h file.
     version_h_lines = [GPL2_HEADER_C_LIKE, GENERATED_HEADER_C_LIKE, '#pragma once']
 
-    version_h_lines.append(f"""
+    version_h_lines.append(
+        f"""
 #define QMK_VERSION "{git_version}"
 #define QMK_BUILDDATE "{current_time}"
 #define CHIBIOS_VERSION "{chibios_version}"
 #define CHIBIOS_CONTRIB_VERSION "{chibios_contrib_version}"
-""")
+#define GIT_COMMIT "{git_commit}{'*' if git_dirty else ''}"
```

Comment:
Maybe `QMK_GIT_HASH` or something so that its partially consistent with other variables, and allows for ChibiOS variants in the future?